torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -45,6 +167,24 @@ void RpcAgent::enableGILProfiling(bool flag) {
   profilingEnabled_ = flag;
 }
 
+steady_clock_time_point RpcAgent::computeNewRpcRetryTime(int retryCount) {
+  // The exponential backoff algorithm being used here is:
+  // newTime = timeNow + (retryDuration * (backoffConstant ^ retryCount)).
+  std::chrono::milliseconds timedelta =
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+          kRpcRetryDuration * pow(kBackoff, retryCount));
+  return std::chrono::time_point_cast<std::chrono::milliseconds>(
+      std::chrono::steady_clock::now() + timedelta);
+}
+
+bool RpcAgent::shouldRetryRpc(std::shared_ptr<FutureMessage> future) {
```

Comment:
The RPC should be retried only if the RPC has completed with an error. Note that every future will eventually be marked as completed via one of these mechanisms:

1. The work completed successfully.
2. Something failed and the future was marked with an error.
3. The RPC timed out and the RpcAgent (ProcessGroup/Thrift) marks the Future with a timed out exception.